---
title: <center>The Animal Model</center>
date: <center>`r format(Sys.Date(),format="%B %d %Y")`</center>
output: 
    html_document:
       css: Chapter.css
       toc: yes
       toc_depth: 5
---


```{r,echo=FALSE,results='hide',include=FALSE}

library(rdryad)
library(kinship2)
library(MCMCglmm)

```

### Background

As a final approach to the problem of understanding the genetics of adaptive traits, we are going to step away from our efforts to identify genes and simply ask whether we can gain insights into the evolution of adaptive traits, even when the kinds of genetic data we have been using heretofore are not available.

### Review

First of all, remember Galton's famous case:

```{r,echo=FALSE}
data(galton)
par(mfrow=c(1,1))
plot(galton$parent,galton$child, pch=19, col="blue", xlab="Parent", ylab="Offspring")
lm1 <-lm(galton$child~galton$parent)
lines(galton$parent,lm1$fitted,col="red",lwd=3)

```
Now supposing the slope of the regression line had in fact been zero.  If so, all of the variance would have been random (what statisticians refer to as the residuals) and we could describe an individual's height as

Y = &mu; + e;


But of course, we did observe a significant slope, which Galton assumed to be due to heritable variation.  Thus, we can expand our model to be

Y = &mu; + a + e

where, in quantitative genetics terms,  a is the "breeding value" of the individual and accounts for the additive contribution of qtl alleles on phenotype.  So our problem is to figure out what fraction of the variance can be accounted for by variance in breeding value; after that, we remember that

h^2^ = Vg/(Vg +Vr)

And we have our estimate. 

But it's not quite that simple.  First, suppose there are other factors that contribute to the phenotype (which in Galton's case there certainly are).  If so, we need to add those components to our model.  Those factors could be **fixed factors** - ones like sex, for which there is no error term associated with the trait of an individual, or they could be **random factors** like, for instance, diet, where there is obviously error associated with any measure we might devise.  So very rapidly the problem becomes more complicated.

But second, what if we have more than just parent-offspring data?  Suppose, for example, we have a pedigree like this:

```{r}
load("../QPopgen/Data2015/Pedigree")
dat.ped <-with(dat.raw,pedigree(id,fatherid,motherid,sex))
plot(dat.ped)
```
In Galton's case, all the relationships involved were the same - the "kinship coefficient", the probability that two alleles drawn at random from two individuals being compared, is the same across the board.  But in this case, the degree of relatedness (kinship, or probability of identity by descent of genes in two individuals), varies.  In fact, we could construct a kinship matrix and look at part of it:
```{r}
kinship(dat.ped)[1:8,1:10]
```

In this simple pedigree, we see that for individuals, K=0.5, meaning that given one allele at a locus, the probability of selecting it a second time is 0.5.  In the case of parent and offspring, the probability of a selected allele being passed to a progeny is 0.5; the probability of it then being selected in the progeny is 0.5, making the probability of the same allele being selected in parent and offspring 0.5^2^, or 0.25.  For parent and grandchild, K=0.5^3^ or 0.125.

So the question becomes, how do we fit our model in such a way that we can estimate the variance components (Va, Vr, and that associated with any other factors included)?.  There are multiple approaches; MCMCglmm, as its name implies, uses a Bayesian approach, in which based on an uniformative set of priors, the sample space for the variance components is explored, giving rise to a posterior distribution for each.

### The data

For this example, we are going to use data published by Maze-Guilmo et al., 2014 (Dryad link [here](http://datadryad.org/resource/doi:10.5061/dryad.s3k34)).  In this work, they looked at a host-parasite system in a wild population of the freshwater fish (*Leuciscus. burdigalensis*, the beaked Dace) in Europe. The parasite, the copepod *Tracheliastes polycolpus*, is a recently introduced one The parasite Specifically, their goal was to assess the heritability of two parasite-related traits:

1.  **Resistance**, in which the host prevents or limits infection, through mechanisms like immune response, behavioral avoidance, etc.
2.  **Tolerance**, in which infection is not limited but its consequences are ameliorated.

These traits can be measured (and you can consult the [original paper](http://rspb.royalsocietypublishing.org/content/281/1779/20132567) for details), but what about establishing relationships (that is, inferring the pedigree)?  Here is the only place in their work in which actual sequence variation was incorporated - in short, they used microsatellite data to infer a pedigree (just as 23andMe is doing when they tell you who your fourth cousins are).  Thus, they could, based on the inferred pedigree, construct a kinship matrix and use it in their analysis.

Their data are available on Dryad; we will use the rdryad package to access them directly.  First the pedigree file
```{r}

#url.ped <-download_url(id='10255/dryad.1759')
#Ped <-dryad_getfile(url.ped)
Ped <-read.table("./Data/dace.pedigree.txt",header=TRUE)
#Ped <-Ped[1:1354,1:3]
head(Ped)
tail(Ped)
```
Note a couple of things.  First, the data as downloaded took some subsetting to get only what we want - the id of each individual as well as that of its parents.  Second, notice that the first set  of individuals (displayed with the head() function) do not have identified parents.  These are the ones that correspond to individuals 1-4 in our pedigree - they are ones for which we have no ancestry information.  They do need to be the first entries in the data.  Contrast them with those at the tail - each individual has an identified father and mother.

And note in passing - there is a lot of uncertainty in this pedigree.  In fact, it is not a single one, but rather a collection of many.  Thus, unlike estimates of heritability made from, for example, animal breeding programs, the kinship matrix has much uncertainty to it.

Now we can obtain the phenotype data:

```{r}
#url.pheno <-download_url(id="10255/dryad.58626")
#Data <-dryad_getfile(url.pheno)
Data <-read.table("./Data/dace.phenotypes.txt",header=TRUE)
names(Data)[1] <- "animal"
attach(Data)
colnames(Data)
```
And we see columns for our two dependent variables we want to consider (Resistance and Tolerance),  , as well as a number of other variables that could be factored into our analysis (year collected, collection site, age, etc.). We can quickly visualize the distributions of the two phenotypes:

```{r}
par(mfrow=c(1,2))
hist(Data$RESISTANCE,main="Resistance",xlab="Level")
hist(Data$TOLERANCE,main="Toleranace", xlab="Level")
var(Data$RESISTANCE)
var(Data$TOLERANCE)
```
So what we will be doing is to be looking at the following relationship

Resistance ~ Size

The reason for using size is that it is a combined measure of age and parasite burden, so it serves as a measure of cumulative parasite exposure.  We will then treat year and station as random factors (with the idea that, while we know what they are, we do not know what environmental variables contribute to differences associated with them).  Finally, we will consider not just their independent effects, but also the effects of any interaction between the two.  

Thus, we can think of our model as something like 

Resistance ~ &mu; Size + a + station + year + station:year + &epsilon;

where a is the additive effects of genes affecting whichever trait we are looking at (Resistance or Tolerance). So we want to estimate the variance components associated with each and then calculate heritability with our familiar formula.  And since we are taking a Bayesian approach, we can estimate 95% credible intervals for each parameter, as well as for h^2^, as well.

### The analysis

#### Establishing the Priors

We won't spend a lot of time on this; suffice it to say that what we are using is a so-called inverse-gamma prior for each variance component, a standard for these analyses (and one that is rather uniformative).  They are generated by the following:

```{r}
prior1.7 <- list(R=list(V=1,nu=0.002),G = list(G1 = list(V = 1, n = 1.002), G2 = list(V = 1,n = 1.002), G3 = list(V =1, n = 1.002),G4 = list(V = 1, n = 1.002)))
```


Not that R is the prior for variance in the response and G1 through G4 are those for the four other  variables in the model.

#### Running the model

The code below runs the model.  There is a lot to it; but not a few things:


1.  Our random factors that affect that relationship are individual animal, year, station and year-station interaction
3. We've provided the identity of the genotype and phenotype data objects, as well as the prior distributions.
4.  We've set the number of MCMC cycles to 100,000, with a 50,000 burnin period.  We've also specified that the data be "thinned" - that is that only every 100th value accepted by included in the final posterior distributions.


```{r,results='hide'}
model1 <- MCMCglmm(RESISTANCE ~ SIZE, random = ~ animal + YEAR + STATION +
YEAR:STATION,pedigree = Ped, data = Data, nitt = 100000, thin = 100, burnin =50000,prior=prior1.7, verbose = TRUE,family="poisson")
```


#### Analyzing the results

Remember above all - what we are interested in are estimates of the variance components, which will lead us to estimates of heritability.  We are also interested in how well our MCMC chains ran, so we want to examine things like the pattern of values obtained as the chains progressed and the effective sample sizes of the posterior distributions.  

We can visualize the results graphically as follows:

```{r}
plot(model1$VCV)
```
On the left are the "traces" - the values for each component added to the posterior as the chain progresses.  Ideally, these should oscillate randomly (and they do).  On the right are the posterior distributions for each of the components - they should be unimodal and (ideally) normally shaped.  In most cases they are (although longer chains would likely improve them.)

But let's get some numbers for these - the mode of each posterior and the 95% credible intervals.

```{r}
pm1 <-posterior.mode(model1$VCV)
pm1
HPDinterval(model1$VCV)
```
And how good is the model at explaining the total variance in resistance?  We can estimate that by taking the sum of the explained variance components (additive, year, station and interaction) and dividing by the total
```{r}
expl1 <-sum(pm1[1:4])/sum(pm1)
expl1
```


To give the components more meaningful labels:

1.  animal - additive genetic variance
2.  units - residual (error) variance

We can also compute effective sample sizes of our distributions:

```{r}
effectiveSize(model1$VCV)
```
And overall this looks good.

Now to calculate heritability, which is simply going to be animal/∑everything:


```{r}
h.model1 <-model1$VCV[,"animal"]/(model1$VCV[,"animal"]+model1$VCV[,"YEAR"]+model1$VCV[,"STATION"]+model1$VCV[,"YEAR:STATION"]+model1$Sol[,"SIZE"])
posterior.mode(h.model1)
HPDinterval(h.model1)
```

And the effective size of our distribution for h^2^ is

```{r}
effectiveSize(h.model1)
plot(h.model1)
```

So, based on the data, we can conclude that there is a narrow sense heritability of resistance is about 0.2.  

#### What about Tolerance?

Remember, we measured two phenotypes - resistance and tolerance.  Thus, we can rerun the analysis above, simply substituting Tolerance for resistance in the model.  We can keep the same priors.


```{r,results='hide'}
model2 <-MCMCglmm(TOLERANCE ~ SIZE, random = ~ animal + YEAR + STATION +
YEAR:STATION,pedigree = Ped, data = Data, nitt = 100000, thin = 100, burnin =50000,prior=prior1.7, verbose = TRUE,family="poisson")
```

After which, we can complete the same analytical steps to verify chain performance and estimate h^2^:
```{r}

pm2 <-posterior.mode(model2$VCV)
pm2
HPDinterval(model2$VCV)
effectiveSize(model2$VCV)
plot(model2$VCV)
```
And the percent explained by the model is
```{r}
sum(pm2[1:4])/sum(pm2)
```


Here, the performance is less robust.  Not in particular that the effective sample size of animal, which is in fact the posterior distribution of the heritable component of variation.  This could likely be rectified by increasing the length of the chain.

And finally, of course the key question is one of heritability:

```{r}

h.model2 <-model2$VCV[,"animal"]/(model2$VCV[,"animal"]+model2$VCV[,"YEAR"]+model2$VCV[,"STATION"]+model2$VCV[,"YEAR:STATION"]+model2$Sol[,"SIZE"])
posterior.mode(h.model2)
HPDinterval(h.model2)
effectiveSize(h.model2)
plot(h.model2)
```
This trait has a value of h^2^ similar to that of resistance, something the authors of the original paper note as one of their most significant findings.  They go on to test whether the two traits are correlated; their conclusion is that they are not.

### Conclusions

In this, we have used the animal model primarily as a way to reinforce the concept of heritability and to show one means whereby it can be estimated in wild populations.  There are, as always, a couple of caveats of which one should be aware:

1.  At the core of any heritability model is a pedigree.  In some cases, it is known; in other cases, such as this one, it is inferred and thus introduces some uncertainty into the conclusions.
2.  As in any model, adding variables to it may improve one's apparent ability to explain observed variation, but one does so at the cost of adding complexity.  Thus, the relative strengths of different models have to be explored carefully.

Finally, of course, while we have heritability estimates for the quantitative traits of resistance and tolerance, absent a genetic map of the species, localizing and identifying qtl's is not practical.  Nevertheless, by taking this approaches and getting estimates of both heritabilities, as well as determining the absence of genetic correlation between the two, we are, at least in theory, construct hypotheses regarding the potential evolutionary trajectories these species might follow given the new selective pressure of this recently introduced parasite.