00---
title: <center>More Aspects of Selection Theory</center>
author: <center>Bruce J Cochrane</center>
date: <center>`r format(Sys.time(), '%d %B, %Y')`</center></br>
output: 
    html_document:
       css: Chapter.css
       toc: yes
       toc_depth: 5
---


### Calculation of fitness from real data

To quickly summarize, we have been focusing on one component of fitness (viability) and modeling the impact of various models on allele frequency dynamics.  In so doing, we have defined relative fitness, mean fitness, and selection coefficients and shown that, by itself, only heterozygote advantage leads to a stable polymorphism (at a cost in mean fitness).  However, other than the hypothetical example we started our discussion with, we have not addressed how we might estimate fitness parameters from real data.  In fact, this is something that is often difficult to do, especially when selection coefficients are small.  However,  it is not impossible.

This example involves the "tired old Bucephalus" of sickle cell anemia and goes from observed genotypes in a large sample of adults to an estimate of relative fitnesses of the three genomes.  The approach is based on observed departure from Hardy Weinberg in those adults **and the assumption that that deviation is the result of viability differences among the three genotypes**.


#### Hardy-Weinberg Calculations

To start with, we can calculate the allele frequencies from the observed adult population and use them to determine the **expected** genotype distribution:

```{r}
genos.n <-c(25374,5482,67)
genos.f <-round(genos.n/sum(genos.n),3)
p <-genos.f[1]+.5*genos.f[2]
genos.exp <-round(c(p^2,2*p*(1-p),(1-p)^2),3)
genos <-rbind(genos.f,genos.exp)
rownames(genos) <- c("Observed","Expected")
round(genos,30)
```

We can then ask if the differences between observed and expected are significant:

```{r}
hw(genos.n)
```

And we see that in fact the chi-squared value is astronomical.  So let's proceed

#### Make some assumptions

In what follows, we make two assumptions:

1.  Mating is random, so **at the time of zygote formation, HWE exists.**
2.  Any observed departure from Hardy-Weinberg ratios in adults is due to egg-to-adult viability differences among genotypes 

#### Calculate relative fitnesses

Given those assumptions there are two ways to proceed.  First, we could assume that there was a pool embryos of some arbitrary size (larger than the number surveyed) in HW equilibrium, calculate the numbers of the genotypes in that pool and then calculate the ratio of observed individuals to those numbers:

```{r}
embryos.n <-genos.exp*40000
w <-round(genos.n/embryos.n,3)
w
```
And in keeping with our norm, we can calculate the values relative to the most fit genotype (which happens to be the heterozygote)

```{r}
w <- round(w/max(w),3)
w
```

But there is actually an easier way to get the same answer.  We have observed frequencies, we also have expected ones, so let's just take the ratio of those two:
```{r}

w2 <-genos.f/genos.exp
w2 <-w2/max(w2)
round(w2,3)
```
And we get essentially the same answer.

#### Summary

This is a classic textbook type of calculation.  However, in addition to the assumptions we've already described, notice the magintude of the selecton coefficients:

```{r}
s <- 1-w[1]
t <-1-w[3]
s;t
```
*Both of these are very large*.  And remember, the calculation is based on observed departure from HW.  In cases where selection coefficients are smaller (as is typical in most cases of polymorphisms), the departure from HW in a reasonable sample size is likely to be insignificant, rendering this sort of calculation suspect at best.




### Adding in mutation

In our discussion thus far, we have used selection as the only factor affecting allele frequencies.  Of course, in an ideal population, we also have to consider mutation.  First, let's consider a recessive detrimental allele.  The loss due to selection will be q^2^ * w(AA), or q^2^ * (1-s).  Since q^2^ &asymp; 0, we can approximate this as -sq^2^.  If the gain due to selection is &mu; , then equilibrium will be reached when

&mu; = sq^2^

or  

q <- &radic; (&mu;/s)

As an example, we can look at some reasonable numbers - µ=10^-7^, s=.01
```{r}
u <-10^-7
s <-.01
q1 <-sqrt(u/s)
q1
```
So at equilibrium, we would expect an allele frequency of `r sqrt(u/s)`, which is small but not insignificant.  We can also look at a range of values of s:

```{r}
par(mfrow=c(1,1))
s <-seq(.002,.1,.001)
q <-sqrt(u/s)
plot(s,q,typ="l")
```

But now let's assume that selection also acts against the heterozygote. We can proceed as follows:

1.  If q is small, then we can treat the number of aa homozgyotes as zero
2.  And in that case, p is nearly 1, so we can approximate the frequency of heterozygotes as 2q.

Thus, the reduction in the number of heterozygotes, using our previously established notation, will be 2qsh; with respect to little a alleles (half of those in the heterozygotes) it will be qsh.  And, of course, once again, the gain due to mutation will be &mu; thus at equilibrium


µ = hsq  

and

q = µ/hs


So let's see what we get, keeping s at .01 but now adding in a weak effect (h=.1) on the heterozygote:

```{r}
h=.1
s <-.01
q2 <-(u/(h*s))
q2
```
And comparing that with what we got above for the purely recessive case:
```{r}
q2/q1
```
So with an effect on heterozygote, equilibrium frequency of p much much lower (ca. two orders of magnitude).  And we can look at that ratio (holding h and µ constant) over a range of values of s, yielding the following

```{r}
s <-seq(.002,.1,.001)
q1 <-sqrt(u/s)
q2 <-u/(h*s)
plot(s,q2/q1,type="l",xlab="s",ylab="ratio")
```
And this makes a biologically important point. The equilibrium frequency of a detrimental allele is much lower when selection acts even weakly on the heterozygote as well as the homozygote.



### Selection in finite populations

Now we need to address the joint effects of seletion and drift.  Based on intuition, we would expect the following.

1.  Large population - drift is small; selection predominates
2.  Small population - drift stronger; case less clear

And before doing so, let's go back about 40 years, to when it first became clear that polymorphism and heterozygosity are common features in populations of most species.  Since simple selection theory, as we've covered thus far, predicts fixation or loss of alleles in most cases, this posed a problem, and indeed led to one of the great controversies in the history of evolutionary biology, the so called "selectionist-neutralist" debate.  While the "debate" has long since lapsed, it is worthwhile to recall the neutralist argument, as it provides a null hypothesis for most experimental tests of natural selection hypotheses.  The major tenets of the theory, as promulgated by Motoo Kimura and others, were simple:

1.  Populations are finite
2.  A large fraction of nucleotide or amino acid substitutions have no effect on fitness.
3.  Therefore, the stochastic processes of mutation and drift are responsible for most observed molecular variation.

So, under this model, levels of polymorphism and heterozygosity are determined largely by &theta;, which is of course 4Neµ.  Indeed Kimura, in his [1971 paper](https://dl.dropboxusercontent.com/u/9752688/ZOO%20422P/Readings/Kimura%201971.pdf) (considered a classic) used this theory to estimate the effective population size of humans, using an estimate of µ of 10^-7^ /year, a generation time of 20 years, and average heterozygosity of .1:

```{r}
Ne=function(H,u){
F <- 1-H
num <- (1/F)-1
denom <-4*u
Ne <-num/denom
Ne
}

as.integer(Ne(.1,10^-7*20))
```

Obviously, there are many assumptions made that add lots of uncertainty to this estimate; we needn't belabor that point.  But there was another problem.  Remember that the "find-em and grind-em" era of electrophoresis was in full swing, so lots of data were being accumulated for lots of species. And here is the pattern that emerged.

```{r}
taxa <-c("Dicots","Molluscs","Crustaceans","Insects","Drosophila","Fish","Amphibians", "Reptiles", "Birds", "Mammals")
H <-c(.05,.145,.08,.09,.12,.05,.07,.055,.05,.04)
Ntax <-sapply(H,Ne,u=10^-7)
barplot(Ntax/10000,names.arg=taxa,las=2,ylab=expression("Ne X 10"^-4))
```

So is it believable that as wide a variety of taxa would all have effective population sizes in a range of &sim;10,000-40,000?  Selectionists argued that this was a serious flaw in the neutral theory - mutation and drift simply could not explain the relative uniformity of heterozygosity levels observed in populations of such disparate organisms.

### Drift and the strength of selection

For our discussion below, we are going to assume the following relative fitnesses (a simple additive model):

w(11) = 1  
w(12) = 1+s  
w(22) = 1+2s  

Now, remember back to our discussion of the fate of a new mutation.  At that time, we saw that, even with a selective advantage, in a finite population, the probability of loss, due simply to the sampling variance inherent in a Wright-Fisher population,is high.  Following along with that logic, it is possible to show that the probability of survival of a new mutation in the first generation is approximately 2s/(1+s)^2^ .  And as s gets small, 1+s approaches 1, so we can approximate this probability as simply 2s.  For example:


```{r}
s <-c(0,.01,.02,.05,.1,.2,.5,1)
p1 <-(2*s)/(1+s)^2
p2 <- 2*s
round(cbind(s,p1,p2),4)
```
But let's take this one step further.  How many mutant alleles would we expect to observe in an infinite population at some time t?  This is simply (1+s)^t^, however this is average of distribution including both those populations that have retained it as well as lost it.  So consider with s=.01 and 1000 generations
```{r}
pret=p1[2] # probability of survival in first generation 
mean.alleles <-(1+.01)^1000 # given survival, number of alleles after 1000 generations
mean.alleles
```
BUT - what is probability that a given population has lost the allele in the first generation?
```{r}
1-pret
```
So our distribution consists of 98% zeros.  If we exclude them, then the average number of alleles is
```{r}
mean.alleles/pret
```

But this assumes N is infinite, which of course it is not. In the context of a finite Ne, what we want to do is to determine the probability of fixation. For a neutral allele, that probability is 4N, but what about when we add in s?  The math is far from trivial, but the key result is in the following equation, which gives the probability of fixation of an allele with a **positive** selection coefficient of s and a population size of N:

![NSeq8.1](http://dl.dropboxusercontent.com/u/9752688/QPopgen/figs/NSeq81.png)

Note that, as in the case of drift, in which the key parameter was 4Neµ, in this case it is 4Nes.   But we can refine this a little bit more and ask how this probability varies with respect to p.  That turns out to be a simple matter of adding 2p to the exponent in the numerator. We can explore that function graphically with the following functions:
```{r}
fix.p <-function(Ns,cl="black"){
p <-seq(0,1,.01)

U <-(1-exp(-4*Ns*p))/(1-exp(-4*Ns))
lines(p,U,col=cl)
}
blank <- function() plot(0,bty='n',pch='',xlab='p',ylab='Prob ability of Fixation', xlim =c(0,1),ylim=c(0,1))
```
First, let's look at the case we did above, in which s=.01 and N varies between 10 and 10,0000
```{r}
blank()
N <-c(10,25,50,100,200,500,1000,10000)
s=.01
Ns <-N*s
x <-sapply(Ns,fix.p)
Ns
```
And we see that as Ns gets large, the probability of fixation of rare allele gets high (and in fact fixation becomes allele frequency-independent).  But note also, these are for positive values of s (i. e. allele makes positive fitness contribution).  What if selection is negative (s<0 in this notation)?
```{r}
blank()
x <-sapply(Ns,fix.p)
Ns <--Ns
x <-sapply(Ns,fix.p, cl="red")
```
And prehaps not surprisingly, we get a mirror imaage of what we saw for positive selection. 

But let's set aside those cases where Ns is large and, regardless of p, probability of fixation or loss is essentially zero. To do so, we can focus on  -.5<Ns<.5

```{r}
blank()
Ns2 <-c(-.5,-.25,-.125,-.1,.1,.125,.25,.5)
x <-sapply(Ns2,fix.p)
abline(0,1,col="red",lty=2)
```

The key point here is that **selection doesn't matter much** - over this range of values, the probability of fixation doesn't vary much from neutral expectations (Ns=0)

[Tomoko Ohta (1976)](http://www.sciencedirect.com/science/article/pii/0040580976900198#) investigated this further by looking at both probability of fixation and, perhaps more importantly, expected heterozygosity, for this case.  In essence, the question she addressed was what are the predicted dynamics of weakly deleterious alleles in finite populations?

First, let's look at fixation probability over the same range as above (-.5 < Ns <.5); furthermore we'll assume that N=500, so 2N=1000, and we are varying s to vary Ns.

```{r}
Ns2 <-c(seq(-1,-.01,.01),seq(.01,1,.01))

U2 <-(1-exp(-4*Ns2*.001))/(1-exp(-4*Ns2))
plot(4*Ns2,U2,type="l", xlab="4Nes",ylab="Fixation Probability",main="Fixation")                           
```

And no big deal here - when selected against, fixation probability is lowered; when positive it is raised.

But remember - our problem is to explain observed levels of heterozygosity. Again, we won't grind through the math,  but we can calculate the ratio of Het(Ns)/Het(neutral) at equilibrium by
```{r}

het <-(2*(4*Ns2-1+exp(-4*Ns2)))/(4*Ns2*(1-exp(-4*Ns2)))

plot(4*Ns2,het,type="l",ylim=c(0,3),ylab="Relative Heterozygosity", xlab="4Nes",main="Heterozygosity")
abline(h=1,col="blue")
```

And to contrast them, let's put them both on the same plot.  
```{r}
plot.new()
par(mfrow=c(1,1))
U2.norm <-U2/.001
plot(4*Ns2,U2.norm,type="l",col="red", main="Fixation Probability and Heterozygosity",xlab="4Ns",ylab="Relative to Neutral")
lines(4*Ns2,het,col="blue")
abline(v=0)
abline(h=1)
```

This figure is worth some careful consideration with respect to what it predicts about weakly selected alleles:

1.If selection is positive, then the fixation rate increases much more quickly than does heterozygosity.  This makes sense, since the stronger positive selection is, the more likely the favored allele will go to p=1 (and heterozygosity = zero).
2.  The more interesting result is in the case of negative selection.  Here we see the oposite kind of relationship - even when the probability of fixation is nearly zero, heterozygosity is comparable to that seen in a purely neutral case.

From these results, therefore, Ohta proposed that heterozygosity of weakly detrimental alleles would be expected to fall in a narrow range.  The upper limit, of course, is determined by selection - if heterozygosity  is high, the allele frequency must be high and thus there are more individuals (especially homozygotes) are the targets of selection.  When heteozygosity is low, on the other hand, selection is inefficient, so that heterozygosity is a function of drift (which is, of course, mediated by population size).

### Conclusions

So where are we at right now?  Keep the following points in mind as we go forward:

1.  In basic selection models, only balancing selection results in stable polymorphisms, and it does so at a cost.
2.  Despite this, observed levels of polymorphism in populations are typically quite high.
3.  While strong selection can be inferred based on departures from Hardy-Weinberg, selection coefficients for most polymorphisms are likely quite small.
4.  If selection coefficients are small, then the joint effects of selection and drift, in theory, place upper and lower bounds on levels of heterozygosity in populations.

All of these, taken together, suggest that inference of the role of selection in shaping patterns of genetic variation in populations is an exceedingly difficult problem.  In fact, what we have developed so far is really insufficient to do so.  But what we haven't taken into account is all of what molecular biology tells us.  In particular, we have been treating all nucleotide substitutions equally.  It's time we develop a more nuanced view of genes and their structure.

<a rel="license" href="http://creativecommons.org/licenses/by-nc-sa/4.0/deed.en_US"><img alt="Creative Commons License" style="border-width:0" src="http://i.creativecommons.org/l/by-nc-sa/4.0/88x31.png" /></a><br />This work is licensed under a <a rel="license" href="http://creativecommons.org/licenses/by-nc-sa/4.0/deed.en_US">Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International License</a>.
