---
title: "BIO 444 Test Questions"
output: html_notebook
---

# 1 Unix Loops
In order to run our migrate-n analyses we had to construct directories, each containing a parameter settings file, and then enter each directory to run migrate. Imagine that instead of 1 species, you had to analyze 40 species with migrate-n.

Write a 'for-loop' in unix that would allow you to start in a directory containing directories with all five hypotheses, move into each directory, start migrate-n, and then move to the next directory until migrate models are running in each directory.

Ignore the need to create virtual terminal windows (screen) for this problem: imagine that migrate-n starts in the background and then lets you issue more commands. To over come this when testing your program, use "echo migrate-n parmfile" as the command to run migrate-n.

You may use the directory linked [here](http://ericcrandall.github.io/lessons/exam/5_hypothesis.zip) to experiment with. Give the code in the box below.

```{bash}
for dir in *.*
  do
  echo dir
  cd dir
  echo migrate-n parmfile
  cd ..
done

```

# 2 Bat R Throw L
The following three questions will be based on this comma-delimited dataset that describes statistics for Major League Baseball in 2008. Please paste in your one line of code, and the resulting R output into the response box.

How many National League baseball players batted right and threw with their left in 2008?

```{r}
bat<-read.table("https://ericcrandall.github.io/BIO444/lessons/exam/batting_players08.txt", header=T, sep=",")

table(bat$bats,bat$throws,bat$lgID)
```
# 3 Names
What are their names? How many runs-batted-in (RBI) did they hit?

```{r}
BRTL<-bat[which(bat$bats=="R" & bat$throws=="L" & bat$lgID=="NL"),c("nameFirst","nameLast","RBI")]
```

# 4 RBI Table
Create a table that is sorted alphabetically by the number of RBIs these players hit in 2008.
```{r}
BRTL[order(BRTL[,3],decreasing=T),]
```

# 5 HW Equilibrium
You measure the frequency of a SNP in a population of corals. You find the following genotypes:
GG: 85
GT: 90
TT: 10

Write code to calculate p, q, the expected genotypic *frequencies* as well as the expected numbers of each genotype if the population were under Hardy-Weinberg, and a $\chi^2$ test (give the $\chi$ statistic as well as the p value) to determine if the population significantly departs from HWE. Provide code as well as results for all of the values requested. Do not use the functions from TeachingPopGen package.

```{r}
obs<-c(85,90,10)
N<-sum(obs)
p <-(2*obs[1]+obs[2])/(2*N)
q <- 1-p
p; q

expF <-c(p^2,2*p*q,q^2)
expN <-N * expF
expN

chi<-sum((obs-expN)^2/expN)
chi

pr<-1-pchisq(chi,1)
pr

```

#6 The General F-statistic

What is going on with the data above? Calculate the generalized F statistic, and from its value, derive a possible explanation. Provide code, your calculated F statistic and a written interpretation.

```{r}
expH<-0.41782323
obsH<-90/N

F<-1-(obsH/expH)

```

#7 The fate of a new mutation

A certain recent mutation in a haploid population confers a 1% increase in reproductive output (s=0.01) to any individual that has it, relative to the ancestral gene, which is neutral. Assuming that an individual with a neutral allele has 1 offspring on average, what is the probability that this brand-new mutation will survive for 200 generations. How does this compare (how much more probable) to a new mutation that did not confer any reproductive advantage?

```{r}
fate <-function(c){
x <-rep(0,200) # create a vector of zeros, which will hold the results of the recursion calculation
for(i in 1:199){
  x[i+1] <-exp((c)*(x[i]-1))
}
return(x)
}

a<-fate(1.01)
b<-fate(1)

pA<-1-a[200]
pB<-1-b[200]

ans<-pA/pB


```
There is a `r pA` chance that this new advantageous mutation will survive. That is `r ans` times more probable than the neutral allele.

# 8 Mutation Models
Which mutational model best describes [this](https://ericcrandall.github.io/BIO444/lessons/exam/testdata.txt) dataset based on the Bayesian Information Criterion? You will need the `phangorn` library (You should be able to read.FASTA directly from the url). What are the important aspects of this model (i.e. what parameters of molecular evolution does it consider?) Show your code and any output from the code, and answer the question. 

Bonus question worth 0 points. What fictional place does `phangorn` refer to?

```{r}

library(phangorn)
data<-read.FASTA("https://ericcrandall.github.io/BIO444/lessons/exam/testdata.txt")
data.phy<-phyDat(data)
data.mt<-modelTest(data.phy,G=T,I=T)

data.mt[order(data.mt$BIC),]
```
# 9
Calculate nucleotide diversity and Tajima's D for [this same dataset](https://ericcrandall.github.io/BIO444/lessons/exam/testdata.txt) https://ericcrandall.github.io/BIO444/lessons/exam/testdata.txt using functions in the `pegas` package. Provide nucleotide diversity values both for `per base` and `per sequence`. Provide a written interpretation of each of these results.

```{r}
library(pegas)
nuc.div(data)
337*nuc.div(data)
tajima.test(data)



```

# 10 What is genetic drift?
What, exactly, is genetic drift? How does it affect genetic diversity within a population? How does it affect genetic diversity between populations?

Sampling error. Decreases within. Increases between.

# 11 Effective Population Size

Define effective population size. How does it differ from census population size?

#12 

The parameter $\Theta$ is very important in population genetics. How is it defined? How can we interpret $\Theta$ parameters in our migrate models?

Calculate $\Theta$ for a locus from the following allele frequencies:
Allele A: 0.345
Allele B: 0.162
Allele C: 0.080
Allele D: 0.080
Allele E: 0.110
Allele F: 0.223

```{r}
freqs<-c(0.345,0.162,0.08,0.08,0.11,0.223)
f.adk<-sum(freqs^2)
th.adk <-(1/f.adk-1)
th.adk
```

#13
You obtain SNP data from 

```{r, echo=F}
library(HardyWeinberg)
dat <-(sapply(c(1:5),function(x) (HWData(nm=1,n=100,f=runif(1,.05,.2),p=runif(1,.1,.9)))))



```

You have 

```{r}