---
title: <h1>DNA Sequence Variation - Introduction</h1>
date: <h2>`r format(Sys.Date(),format="%B %d %Y")`</h2>

csl:  ../genetics.csl.txt
output:
  html_document:
    code_fold: show
    css: ../Chapter.css
    toc: yes
    toc_depth: 5
---


## DNA Sequence Variation

The most basic level of genetic variation is, of course, that of DNA sequence.  

### Some data

As an example, we are going to use data from the acp29 locus in *Drosophila melanogaster*,  Popset [11245776](http://www.ncbi.nlm.nih.gov/popset/11245776), downloaded as a FASTA file.  We will use the `rentrez` package to access the data directly:

```{r,eval=T,echo=FALSE, message=F}
library(rentrez)
library(ape)
library(pegas)
library(strataG)
```

```{r}
dat.srch <-entrez_search("popset",term="11245776")
dat <-entrez_fetch("popset",id=dat.srch$ids,rettype="fasta",retmode=xml)
write(dat,file="../Data/acp29.fasta")
acp29 <-read.FASTA("../Data/acp29.fasta")
acp29
```


Note that at this point, we have saved the downloaded data as a local file, which we will be able to access in the future.

We see that there are 17 sequences, each 702 base pairs in length.  We're going to do two processing steps, one trivial and one critical:  

1. As a cleanup for now, we will rename them as simply 1 through 17.  
2.  We want to be sure they are properly aligned.  To do so, we will use the add-on program `muscle` (which must be indepenently installed so that it can be called from R).

```{r}
names(acp29) <-c(1:17)
acp.mat <-muscle(acp29) # align the data with muscle

#or alternatively, read it in from here
#acp.mat<-read.dna("https://ericcrandall.github.io/BIO444/lessons/Data/acp29_aligned.fasta",format="fasta")

```

Note that our data (acp.mat) is now a DNAbin object, which is simply a matrix of sequences - one row per sequence and one column per position.  This is a data type used extensively in ape and related packages. 

Now, we need to visualize them, for a couple of reasons.  First, we need to check our alignment, so that, in doing our analyses, we are as certain as we can be that we are comparing evolutionarily homologous positions.  Second, we need to see, at least qualitatively, how variable the sequences are.  

To do this, we will use the image.DNAbin function, part of the package "ape"
```{r}
image.DNAbin(acp.mat,cex=.8)
```
And with this, we can, at least qualitatively, address our question.  The fact that most bases are identical in all 17 sequences tells us that the sequence alignment looks good; the fact that there are some exceptions to this says that there is in fact variation.

But how much variation?  There are two measures that we will use extensively for this:

#### Number of segregating sites

Since we are interested in variation, our attention should be focused on those sites that are in fact polymorphic (they are therefore Single Nucleotide Polymorphisms).  The ape package provides a function to give us those sites:

```{r}
acp.sites <-seg.sites(acp29)
acp.sites
```

And we can count them rather easily

```{r}
acp.nss <-length(acp.sites)
acp.nss
```

And we see there are 15 out of the total of 702 positions that show some degree of variation.  These are what we need to focus on - from a population perspective the remaining 687 sites tell us nothing.

#### Average pairwise differences

OK, so we have 15 segregating sites.  But there is another question we need to ask - on average, how many differences are there between two sequences?

Let's choose a couple of sequences at random and see how many differences there are between them.

```{r}

seqs <-sample(1:17,2)
seqs
```

Now let's see how many segregating sites there are.

```{r}
ss <-length(seg.sites(acp.mat[seqs,]))
ss
```
And we see there are `r ss`.  If we reran the above block of code, we could select two other sequences, and we'd get another number.  In fact, there are n(n-1)/2 such comparisons; the average of them is our measure of diversity.  
$$\pi =  \sum_{ij}x_ix_j\pi_{ij}$$
Where $x_i$ and $x_j$ are the respective frequencies of the ith and jth sequences in the sample and $\pi_{ij}$ is the number of nucleotide differences between these two sequences.

Again, there is an R function (in the package pegas) that gets at this

```{r}
acp.div <-nuc.div(acp.mat)
acp.div
```
But this is an important point to note.  What we see here is the average diversity *per nucleotide*.  In other words, if we were to pick a random position from two random sequences, this would be the probability that they would be different bases.  We will use this occasionally, but for now, something that will be much more useful will be Ï€, or the average number of differences *per sequence*.  We can get this simply by multiplying the diversity number by the length of the sequence (702 in this case).

```{r}
acp.pi <-ncol(acp.mat)*acp.div
acp.pi
```

### Haplotype Diversity

A final question we can ask is: how unique is any given "haplotype" in this dataset? A haplotype (think "haploid genotype") here is simply the string of bases that are found on one chromosome, but if we had data from multiple loci, it would still be all of the bases found on one chromosome.

```{r}
acp.hap <-haplotype(acp.mat)
acp.hap
```

So among the 17 sequences in the data set, there are 10 different haplotypes. Now we can calculate haplotype diversity as:

$$h = \frac{N}{N-1}(1 - \sum_{i=1}^n X_i)$$

Where $X_i$ is the relative haplotype frequency of each haplotype in the sample, and N is sample size. There is a function in the strataG package to calculate h, but we will write one ourselves

```{r}


hap.diversity<-function (x){ 
  #adapted from diversity() in StrataG
  if (!(is.vector(x) | is.factor(x))) 
    stop("'x' must be a character or numeric vector, or a factor")
  x <- na.omit(x)
  n <- length(x)
  x.freq <- prop.table(table(x))
  out<- (n/(n-1))*(1- sum(x.freq^2))
  return(out)
}

acp.hap.labs<-labelHaplotypes(acp.mat)$haps
acp.h<-hap.diversity(acp.hap.labs)
acp.h


```

Let's check ourselves before we wreck ourselves:

```{r}
library(strataG)
acp29g<-sequence2gtypes(acp.mat)
acp29g<-labelHaplotypes(acp29g)
acp29
```

### Summary

In this brief introduction, we have seen how we can quantify DNA sequence variation based on estimates of three parameters - the number of segregating sites, the average pairwise difference between sites, and haplotype diversity.  To an extent, these are analogous to the measures used in the electrophoresis days - the fraction of loci that are polymorphic and average heterozygosity.  However, as we shall see, there is much more that can be learned based on sequence variation, especially when it comes to making inferences about past evolutionary processes.  To get there, we have to consider a few more factors:

1.  What are the processes that underly generation of variation?  In other words, how can we model the mutational process?
2.  What are the dynamics of the interaction between mutation and finite population size?
3.  How can we make inferences about the processes that gave rise to an observed pattern of sequence variation?

